vsintegration/src/FSharp.Editor/CodeLens/CodeLens.fs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace rec Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Windows.Controls
+open System.Windows.Media
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Editor
+open Microsoft.VisualStudio.Text.Formatting
+open System.ComponentModel.Composition
+open Microsoft.VisualStudio.Utilities
+open Microsoft.CodeAnalysis
+
+type internal CodeLensAdornment(view:IWpfTextView) as this=
```

Comment:
@Pilchie C# Code Lens does not allow colored text, doesn't it? We'd like to color type sygnatures, so we need to implement an adornment ourselves anyway.